<launch>
  <!-- robot_description must be on the parameter server -->

  <node pkg="hand_kinematics" name="ff_kinematics" type="hand_kinematics_coupling" output="screen">
    <param name="root_name" type="string" value="rh_palm"/>
    <param name="tip_name" type="string" value="rh_fftip"/>
    <param name="maxIterations" type="int" value="1000"/>
    <param name="epsilon" type="double" value="0.001"/>
    <!--		<param name="lambda" type="double" value="0.02"/>-->

  </node>
  <node pkg="hand_kinematics" name="mf_kinematics" type="hand_kinematics_coupling" output="screen">
    <param name="root_name" type="string" value="rh_palm"/>
    <param name="tip_name" type="string" value="rh_mftip"/>
    <param name="maxIterations" type="int" value="1000"/>
    <param name="epsilon" type="double" value="0.001"/>
  </node>
  <node pkg="hand_kinematics" name="rf_kinematics" type="hand_kinematics_coupling" output="screen">
    <param name="root_name" type="string" value="rh_palm"/>
    <param name="tip_name" type="string" value="rh_rftip"/>
    <param name="maxIterations" type="int" value="1000"/>
    <param name="epsilon" type="double" value="0.001"/>
  </node>
  <node pkg="hand_kinematics" name="lf_kinematics" type="hand_kinematics_coupling" output="screen">
    <param name="root_name" type="string" value="rh_palm"/>
    <param name="tip_name" type="string" value="rh_lftip"/>
    <param name="maxIterations" type="int" value="1000"/>
    <param name="epsilon" type="double" value="0.001"/>
  </node>
  <node pkg="hand_kinematics" name="th_kinematics" type="hand_kinematics_coupling" output="screen">
    <param name="root_name" type="string" value="rh_palm"/>
    <param name="tip_name" type="string" value="rh_thtip"/>
    <param name="maxIterations" type="int" value="1000"/>
    <param name="epsilon" type="double" value="0.001"/>
  </node>
</launch>


